electron/app/components/RangeSlider.tsx

```diff
@@ -0,0 +1,198 @@
+};
+
+const RangeSlider = ({ rangeAtom, boundsAtom }: Props) => {
+  const [value, setValue] = useRecoilState<Range>(rangeAtom);
+  const bounds = useRecoilValue<Range>(boundsAtom);
+  const [localValue, setLocalValue] = useState<Range>([null, null]);
+  useEffect(() => {
+    JSON.stringify(value) !== JSON.stringify(localValue) &&
+      setLocalValue(value);
+  }, [value]);
+
+  const hasBounds =
+    bounds.every((b) => b !== null) && bounds[1] - bounds[0] > 0;
+  const hasValue = value.every((v) => v !== null);
+
+  return hasBounds && hasValue ? (
+    <SliderContainer>
+      {bounds[0].toFixed(2)}
+      <Slider
+        value={[...localValue]}
+        onChange={(_, v: Range) => setLocalValue([...v])}
+        onChangeCommitted={(_, v: Range) => {
+          setLocalValue([...v]);
+          setValue([...v]);
+        }}
+        classes={{
+          thumb: "thumb",
+          track: "track",
+          rail: "rail",
+          active: "active",
+          valueLabel: "valueLabel",
+        }}
+        aria-labelledby="range-slider"
+        getAriaValueText={valueText}
+        valueLabelDisplay={"on"}
+        max={bounds[1]}
+        min={bounds[0]}
+        step={(bounds[1] - bounds[0]) / 100}
+      />
+      {bounds[1].toFixed(2)}
+    </SliderContainer>
+  ) : null;
+};
+
+const NamedRangeSliderContainer = styled.div`
+  margin: 6px;
+`;
+
+const NamedRangeSliderHeader = styled.div`
+  display: flex;
+  justify-content: space-between;
+`;
+
+const RangeSliderContainer = styled.div`
+  background: ${({ theme }) => theme.backgroundDark};
+  box-shadow: 0 8px 15px 0 rgba(0, 0, 0, 0.43);
+  border: 1px solid #191c1f;
+  border-radius: 2px;
+  margin-top: 0.5rem;
+  color: ${({ theme }) => theme.fontDark};
+`;
+
+type NamedProps = {
+  rangeAtom: RecoilState<Range>;
+  boundsAtom: RecoilState<Range>;
+  includeNoneAtom: RecoilState<boolean>;
+  name: string;
+  valueName: string;
+  color: string;
+};
+
+export const NamedRangeSlider = ({
+  color,
+  name,
+  valueName,
+  includeNoneAtom,
+  ...rangeSliderProps
+}: NamedProps) => {
+  const [includeNone, setIncludeNone] = useRecoilState(includeNoneAtom);
+  const [range, setRange] = useRecoilState(rangeSliderProps.rangeAtom);
+  const bounds = useRecoilValue(rangeSliderProps.boundsAtom);
+
+  const isDefaultRange = range[0] === bounds[0] && range[1] === bounds[1];
+  const hasBounds = bounds.every((b) => b !== null);
+  return (
+    <NamedRangeSliderContainer>
+      <NamedRangeSliderHeader>
+        Confidence{" "}
+        {!isDefaultRange || !includeNone ? (
+          <a
+            style={{ cursor: "pointer", textDecoration: "underline" }}
+            onClick={() => {
+              setRange([...bounds]);
+              setIncludeNone(true);
+            }}
+          >
+            reset
+          </a>
+        ) : null}
+      </NamedRangeSliderHeader>
+      <RangeSliderContainer>
+        {hasBounds && <RangeSlider {...rangeSliderProps} />}
```

Comment:
This slider isn't showing up in a test dataset where only half of objects have confidences (with all objects under the same field). I think this check might be related. I tried changing the two `.every()` calls above to `.some()`, but that didn't seem to help.